src/main/java/com/labs/introtoprogramming/lab4/image/io/bmp/BMPParser.java

```diff
@@ -0,0 +1,66 @@
+package com.labs.introtoprogramming.lab4.image.io.bmp;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Arrays;
+
+class BMPParser {
+
+  int parseHeader(byte[] headerInfo) throws UnsupportedFileFormat {
+    if (headerInfo.length != 14) {
+      throw new UnsupportedFileFormat("File format is not a BMP");
```

Comment:
Use "data"  instead of "file".
It is better to use some other exception here, as data is not always read from file. Image may be loaded via network, or some other byte array input stream.